taikoxyz/taiko-client
# feat(prover): Auction

todo:
- discuss and implement multiple proper `BiddingStrategy`
- review all logic
- tests
- ensure `ProofWindow` on the `Bid`, which should be at least > than `state.avgProofTime * config.auctionProofWindowMultiplier`
- ensure client is periodically calling `isBlockProvableBy` for various blocks to see if the proofWindow has elapsed and they can start generating proofs for auctions that have elapsed, plus flags to configure if this behavior is desired
- check `onBlockVerified` if a new auction is available
